accepted/future-releases/extension-types/feature-specification.md

```diff
@@ -0,0 +1,1212 @@
+# Extension Types
+
+Author: eernst@google.com
+
+Status: Accepted.
+
```

Comment:
Can we start with a motivating example, similar to https://github.com/dart-lang/language/blob/main/accepted/future-releases/inline-classes/feature-specification.md#motivation ?

Reply:
(I'm slighlty split here. Motivating examples are nice for getting things accepted, but once accepted, I'd like a tight, sharp specification the of what the feature does, or discussions about alternatives. Just the facts.

Could we split it in two files? Motivation/discussion and specification. If the specification doesn't work without the other text, then it's already non self-contained enough.